cmd/trace-agent/subcommands/run/command.go

```diff
@@ -49,40 +41,25 @@ func setParamFlags(cmd *cobra.Command, cliParams *RunParams) {
 	cmd.PersistentFlags().StringVarP(&cliParams.CPUProfile, "cpu-profile", "l", "",
 		"enables CPU profiling and specifies profile path.")
 	cmd.PersistentFlags().StringVarP(&cliParams.MemProfile, "mem-profile", "m", "",
-		"enables memory profiling and specifies profilh.")
+		"enables memory profiling and specifies profile.")
 
 	setOSSpecificParamFlags(cmd, cliParams)
 }
 
-func runFx(ctx context.Context, cliParams *RunParams, defaultConfPath string) error {
+func runFx(cliParams *RunParams, defaultConfPath string) error {
 	if cliParams.ConfPath == "" {
 		cliParams.ConfPath = defaultConfPath
 	}
-	return fxutil.OneShot(Run,
-		fx.Supply(&contextSupplier{ctx: ctx}),
-		fx.Supply(cliParams),
-		config.Module,
+	return fxutil.Run(
 		fx.Supply(coreconfig.NewAgentParamsWithSecrets(cliParams.ConfPath)),
 		coreconfig.Module,
+		config.Module,
+		// Required to avoid cyclic imports.
+		fx.Supply(agent.Params{
+			CPUProfile:  cliParams.CPUProfile,
+			MemProfile:  cliParams.MemProfile,
+			PIDFilePath: cliParams.PIDFilePath,
+		}),
+		trace.Bundle,
```

Comment:
@truthbk @hush-hush @ogaca-dd After some debugging, I found out that my component's `newAgent` and `start` method are never called.

My guess is that I'm not doing properly the injection here. Right now, when the agent starts it blocks on `fxutil.Run` waiting on `app.Done()`.

That's why it doesn't show any logs, causing `TestCreditCards` to fail because it looks for a specific text in the agent logs and timeouts after 10 seconds. This test has no relation with the agent itself, but it's part of the integration tests. They build the trace-agent on the fly and run the binary, so they are acting as smoke tests in the current situation.

I'd appreciate your feedback. Thanks!

Reply:
Your component is created only if it is used. As your component is defined as an [empty interface](https://github.com/DataDog/datadog-agent/pull/19466/files#diff-6699922c80e372c638ad4655da617024d3e81f35f360d876f6901cc208f949eeR17) you have to force the creation of your component. You can do that by adding [here](https://github.com/DataDog/datadog-agent/pull/19466/files#diff-6699922c80e372c638ad4655da617024d3e81f35f360d876f6901cc208f949eeR21)  something like `fx.Invoke(_  agent.Component) {}`